ChapmanUniversity1.0/Controllers/SemestersController.cs

```diff
@@ -53,27 +52,45 @@ public IActionResult Create(Semester semester)
             List<string> seasonsList = new List<string>(Enum.GetNames(typeof(Seasons)));
             ViewData["Seasons"] = new SelectList(seasonsList);
 
-            var course = _unitOfWork.Courses.GetById(semester.CourseId);
-            var semesterExists = _unitOfWork.Semesters.SemesterExists(course.Id, semester.CourseSeason);
+            var semesters = _unitOfWork.SemesterRepository.Get();
 
-            if (!semesterExists)
+            var enumerable = semesters as Semester[] ?? semesters.ToArray();
+
+            if (!enumerable.Any())
             {
                 Semester newSemester = new Semester()
                 {
                     CourseId = semester.CourseId,
                     CourseSeason = semester.CourseSeason
                 };
-                
-                _unitOfWork.Semesters.Add(newSemester);
+
+                _unitOfWork.SemesterRepository.Add(newSemester);
                 _unitOfWork.Complete();
+
+                return RedirectToAction(nameof(Create));
             }
-            
+            foreach (var row in enumerable)
+            {
+                if (row != semester)
+                {
+                    Semester newSemester = new Semester()
+                    {
+                        CourseId = semester.CourseId,
+                        CourseSeason = semester.CourseSeason
+                    };
+
+                    _unitOfWork.SemesterRepository.Add(newSemester);
+                    _unitOfWork.Complete();
+
+                    return RedirectToAction(nameof(Create));
+                }
+            } 
```

Comment:
Same thing here with the 2 conditions. nest the conditions if you have to and also try splitting this out into another class.